TM1py/Services/CellService.py

```diff
@@ -437,10 +501,11 @@ def write_through_unbound_process(self, cube_name: str, cellset_as_dict: Dict, i
             # use default 'Numeric' so that not existing elements trigger minor error during TI execution
             element_type = measure_dimension_elements.get(coordinates[-1], 'Numeric')
 
-            if element_type == 'String':
+            if element_type == 'String' or isinstance(value, str):
```

Comment:
Why do we check with `isinstance` ?
Shouldn't the element type of the measure element determine whether it's CellPutN or CellPutS exclusively?

Reply:
This was an interesting one. In my test I was writing back the word "Test" into a string cell. On the failure test case set the cellset to use an invalid measure, "Amounts" instead of "Amount". Then when it tries to retrieve the type of that element it can't, and instead returns numeric. Therefore the TI statements take the form "CellPutN(Test,Dim1,Dim2)". This statement results in a very odd TI error that says it can't find the variable "Test", because it is not wrapped in single quotes. By checking the instance of the value the TI reads "CellPutS('Test', Dim1, Dim2)" which returns a much clearer error KeyError can't find element Amounts in the dimension measures.

Reply:
OK. Makes sense. Then let's leave it like that :)

Reply:
The change is harmful if numbers come in scientific notation. We must rely exclusively on the type of the measure element.
See this test: https://github.com/cubewise-code/tm1py/blob/a44842d5012f61d970f2b1b8d873e417e32184fe/Tests/CellService.py#L440